src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/overloadmethodsdeclarationorder/Example2.txt

```diff
@@ -12,4 +12,6 @@ public void foo(String s) {} // OK
 public void notFoo() {} // violation. Have to be after foo(String s, int i)
 public void foo(int i, String s) {}
 public void foo(String s, int i) {}
+private interface Testing() {}
+public void foo(double d) {} // violation. Have to be after foo(String s, int i)
```

Comment:
All examples should be same by structure , please add such lines to Example1 in correct order/grouping.

Reply:
> this looks like false positive

I analyzed the report, there is a semicolon between the method it is giving error for and its above method.

```java
    public void visit(Node.IncludeAction n) throws JasperException {
        JspUtil.checkAttributes("Include action", n, includeActionAttrs,
                err);
        n.setPage(getJspAttribute(null, "page", null, null, n
                        .getAttributeValue("page"), java.lang.String.class, n,
                false));
        visitBody(n);
    };

    public void visit(Node.ForwardAction n) throws JasperException { // violation
        JspUtil.checkAttributes("Forward", n, forwardActionAttrs, err);
        n.setPage(getJspAttribute(null, "page", null, null, n
                        .getAttributeValue("page"), java.lang.String.class, n,
                false));
        visitBody(n);
    }
```

Here you can see the semicolon at the end of the first method which is meaningless and causing the error. So it is not a false positive and our check is working fine. @romani

Reply:
> All examples should be same by structure , please add such lines to Example1 in correct order/grouping.

done.

Reply:
> Should we change message ?

I guess we should also change the module name as it contains "order" but we're checking the grouping of the methods not the ordering. The new name can be something like [ConstructorsDeclarationGrouping](https://github.com/checkstyle/checkstyle/issues/14726). What do you think about this?

Reply:
> Here you can see the semicolon at the end of the first method which is meaningless and causing the error. So it is not a false positive and our check is working fine

@romani ping

Reply:
>Here you can see the semicolon at the end of the first method which is meaningless and causing the error.

uhh, it is hard to catch but, Check is right. `;` is empty statement
Please add this case to your Inputs, and make a comment over it to explain.

Reply:
@romani ping